### PR TITLE
2025 deal with intermittent upload error

### PR DESCRIPTION
**Pull request by cosa65**
# Description

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2025

#### Link to staging deployment URL (or set N/A)
https://ui-martinfosco-ui7791.scp-staging.biomage.net/

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [x] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

**Comment by cosa65** [comment]
From what I have seen, the origin of the error seems to be related to a chrome caching issue in the 2 calls (`OPTIONS` and `PUT`) involved with doing the axios PUT request to s3. 
This triggers a CORS error because chrome changes the headers being sent between these 2 calls.

File: src/redux/actions/samples/createSampleFile.js [hunk 1]
```diff
@@ -41,20 +55,9 @@ const createSampleFile = (
...
-        fileName: fileNameForApiV1[type],
-        fileDiff: fileForApiV1,
-      },
-    });
-
     return signedUrl;
   } catch (e) {
-    // Can't update the upload status because we didn't even get to create the sample file
-    handleError(e, endUserMessages.ERROR_BEGIN_SAMPLE_FILE_UPLOAD);
+    dispatch(updateSampleFileUpload(experimentId, sampleId, type, UploadStatus.UPLOAD_ERROR));
```
Comment (StefanBabukov) [inline src/redux/actions/samples/createSampleFile.js:60]: So if it fails you try to upload it again and if that fails again then the error message is outputted to the user? 
If thats the case why dont you have ` dispatch(updateSampleFileUpload(experimentId, sampleId, type, UploadStatus.UPLOAD_ERROR));` in this try too?

Comment (cosa65) [reply src/redux/actions/samples/createSampleFile.js:60]: That's the change, I replaced the the push notification with the dispatch upload error

Comment (cosa65) [reply src/redux/actions/samples/createSampleFile.js:60]: I don't understand exactly the suggestion

Comment (StefanBabukov) [reply src/redux/actions/samples/createSampleFile.js:60]: Oh I thought the two actions createsamplefile and updatesamplefile could be one, but I saw that there is a difference

**Review by StefanBabukov — APPROVED** [review]

**Pull request merged**